### PR TITLE
Feature/unified comments paging and states

### PR DESCRIPTION
**Pull request by khaykov**
This PR adds simulated paging and loading states to the Unified Comment list.

Currently, you can test the error behavior by enabling airplane mode. 

It's a bit difficult to test errors without network/db layer, so there is one instance where list behavior is temporarily weird - if you load comments, enter airplane mode, perform pull-to-refresh, and rotate a device - the list will be empty. This is expected.

About differences between our current approach to similar functionality and how it works with Paging 3:
- While the data from endpoint/db passes through the VM, it is only there for a purpose of transformation and addition elements, like sub-headers. VM does not know about the data in the list and does not hold it.
- Because of this, loading/error states of the list are managed internally inside PagingAdapter and delivered to us. I pass those states to the VM and transform them into a UiState.
- Adapter is also responsible for requesting data refresh and retry. VM is not involved in this, and only reacts to the result of this action when the loading/error state of the list changes.
- Paging loading indicator and retry button are managed by a separate `LoadingStateAdapter` adapter that you combine with `PagingDataAdapter`. While loading/error state of paging are delivered to us, we don't need to react to them, unless we want to show a SnackBar or something similiar.

[Image from Gyazo](https://gyazo.com/4fb2c6b657c2df4e65bad2ab050bef77)

[Image from Gyazo](https://gyazo.com/bcf0dd1984bd3fc521ca1b109c88a631)

To test:
- Navigate to unified comments.
- Confirm that the initial loading indicator is visible.
- Scroll down the list, confirm that the paging loading indicator is visible (we do preloading, btw!).
- Turn on airplane mode.
- Perform pull-to-refresh and confirm that the Snackbar with the error message is visible.
- Scroll the list down to activate paging, confirm that the retry button and error toast is visible.
- Pressing on the Retry button after disabling airplane mode should continue the load.
- While the airplane mode is enabled, reopen Unified Comments.
- Confirm that the full-screen error message is visible after the initial load fails.

## Regression Notes
1. Potential unintended areas of impact
Functionality is still behind a feature flag.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/comments/unified/PagedListLoadingState.kt [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    data class Error(val throwable: Throwable) : PagedListLoadingState()
+    data class EmptyError(val throwable: Throwable) : PagedListLoadingState()
+}
+
+fun CombinedLoadStates.toPagedListLoadingState(hasContent: Boolean): PagedListLoadingState {
+    val isLoading = refresh is Loading && !hasContent
+    val isRefreshing = refresh is Loading && hasContent
+    val isNothingToShow = refresh is NotLoading && append.endOfPaginationReached && !hasContent
+    val isError = refresh is Error
+    val isPagingError = append is Error
```
Comment (develric) [inline WordPress/src/main/java/org/wordpress/android/ui/comments/unified/PagedListLoadingState.kt:26]: If all this Boolean are expected to be mutually exclusive, I would consider to use something like an enum maybe. wdyt?

**Review by develric — APPROVED** [review]
Hi @khaykov 👋 ! Code LGTM, good job leveraging StateFlow/SharedFlow and all around 👍 ! Testing also worked as described ! Only left a minor comment and as shared on slack just noting that it feels a bit strange to my eyes to have the list of comments sliding up on snackbar appearing. Both this are not blocking though, plus this all is behind feature flag so we can address in next PRs eventually. Currently there is an issue with connected tests that should be fixed soon. Feel free to merge this once all is green in develop 🙇 !

File: WordPress/src/main/java/org/wordpress/android/ui/comments/unified/PagedListLoadingState.kt (hunk 1 above)
Comment (khaykov) [reply WordPress/src/main/java/org/wordpress/android/ui/comments/unified/PagedListLoadingState.kt:26]: Sorry, I got what you are saying (multiple booleans potentially could be true at the same time), but I'm not really sure how to solve this with enum?

**Comment by AliSoftware** [comment]
I'm moving this to milestone 17.9 since I'm doing code freeze for 17.8 but this PR is still pending some feedback to be addressed and also has conflicts so is not ready to be merged before 17.8 code freeze; so it will make it to 17.9 instead.

Please ping me and your EW if you feel it's really important to make it to 17.8 and we could consider retargetting it to the beta if it's really important.

**Comment by AliSoftware** [comment]
Seems this haven't moved during last sprint and is still in conflict, so moving its milestone again (to 18.0) since I'm doing 17.9 code freeze right now.

**Comment by khaykov** [comment]
We moved to a different approach with Paging, so I'm closing this one in favor of a separate PR.

**Pull request closed without merging**